### ISSUE TITLE
Set up Simulation environment

### ISSUE THREAD

**Issue by KillickNZ** [New-Zealand-Zero/PowerMaker#22]
We need to set up a simulation environment to handle when we test out new strategies.

@MalawiCoffee are you able to give a description of what this looks like from a technical point of view?

What things need to be in place?
What data do we need?
How would you suggest it be done?

**Comment by timothyjosull1** [comment]
@KillickNZ I've coincidentally started setting this up this morning, we should have a discussion between us three, not far off from a working prototype.

**Comment by KillickNZ** [comment]
Awesome! What a machine. Yea keen for that - I can likely find some time on Sunday arvo?

**Comment by MalawiCoffee** [comment]
Hey

I was thinking just simulating what your daily generation and storage looks like. Then matching that to what the spot market was doing that day. That way you can adjust those variables to see how it would work for different locations. You will definitely need to know how much you can inject, fuse size or transformer size. Not entirely sure the best approach. Could keep it simple just basic numbers or we could go very in-depth modelling the lines transformer etc.

Cheers,

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Sebastian Chapman ***@***.***>
Sent: Thursday, June 6, 2024 10:13:15 AM
To: New-Zealand-Zero/PowerMaker ***@***.***>
Cc: Hugo ***@***.***>; Mention ***@***.***>
Subject: [New-Zealand-Zero/PowerMaker] Set up Simulation environment (Issue #22)


We need to set up a simulation environment to handle when we test out new strategies.

@MalawiCoffee<https://github.com/MalawiCoffee> are you able to give a description of what this looks like from a technical point of view?

What things need to be in place?
What data do we need?
How would you suggest it be done?

**Comment by timothyjosull1** [comment]
Agreed Hugo, we can add dummy information for other variables when we have them, e.g., historic solar production (this one's good for estimating https://pvwatts.nrel.gov/), historic uncontrollable load profiles, etc. Comparison of just a battery and spot price is a really good place to start though, other variables tend to be much more predictable.

**Comment by MalawiCoffee** [comment]
I just downloaded hourly from that site. Seems pretty good for simulation data. It would be nice to get it down to a 5 minute level. I’m just playing around with making a dataset showing spot market and solar generation in one. I need to add some battery storage and export capacities. If you have any other sources you know of throw them, my way. I will have a tinker.

Cheers

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by timothyjosull1** [comment]
Sounds good. Historically I've just interpolated it down to 5 minutes, which is not ideal but I havent managed to find an alternative  source. 

Regarding a data set, I've made a script "get_historical_spot_prices.py" which pulls the node's RTD prices from EMI over a selected data range and saves it to the PM database that the other scripts make. It's in the "add-simulation-environment" branch. You could add something to it which reads data from the PVWatts API and you'll have your spot and solar together.

**Comment by KillickNZ** [comment]
Hey guys just catching up on all of this now! Looks like you lads have been busy! Is there anything you need from my end?

**Comment by timothyjosull1** [comment]
Hey @KillickNZ, sometimes the units weren't clear in the functions so I had to make some assumptions in the functions I've built. It would be good at some stage to run through these and make sure all my unit conversions are starting from the right spot

**Comment by KillickNZ** [comment]
Hey mate - I will be up in Queenstown this evening if you wanted to go over things? And will be staying at Forest Lodge all week if you wanted to come over and have a look at how everything is set up.

Cheers
